frame/support/procedural/src/storage/instance_trait.rs

```diff
@@ -70,42 +69,40 @@ pub fn decl_and_impl(scrate: &TokenStream, def: &DeclStorageDefExt) -> TokenStre
 		.and_then(|i| i.instance_default.as_ref())
 	{
 		impls.extend(quote! {
+			/// Hidden instance generated to be internally used when module is used without
+			/// instance.
 			#[doc(hidden)]
 			pub type #inherent_instance = #default_instance;
 		});
 	} else {
 		let instance_def = InstanceDef {
 			prefix: String::new(),
 			instance_struct: inherent_instance,
-			doc: quote!(#[doc(hidden)]),
+			doc: quote!(
+				/// Hidden instance generated to be internally used when module is used without
+				/// instance.
+				#[doc(hidden)]
+			),
 		};
 		impls.extend(create_and_impl_instance_struct(scrate, &instance_def, def));
 	}
 
 	impls
 }
 
-fn create_instance_trait(
+fn reexport_instance_trait(
+	scrate: &TokenStream,
 	def: &DeclStorageDefExt,
 ) -> TokenStream {
-	let instance_trait = def.module_instance.as_ref().map(|i| i.instance_trait.clone())
-		.unwrap_or_else(|| syn::Ident::new(DEFAULT_INSTANTIABLE_TRAIT_NAME, Span::call_site()));
-
-	let optional_hide = if def.module_instance.is_some() {
-		quote!()
+	if let Some(i) = def.module_instance.as_ref() {
+		let instance_trait = &i.instance_trait;
+		quote!(
+			/// Reexport from support::traits.
+			// This reexport is not strictly needed but used in order not to have breaking change.
+			pub use #scrate::traits::Instance as #instance_trait;
```

Comment:
there is two reexport here: local reexport and public reexport.

Indeed if we don't mind breaking public reexport we can do `use #scrate::traits::Instance as Instance;` (e.g. nobody would have included the trait somewhere), I'm ok with this.

But for local reexport all instantiable crate currently use it.
E.g. when in pallet-balance we do `pub trait Trait<I: Instance = DefaultInstance>` we use local reexport.

Are you ok with only local use ?

```suggestion
			/// Local import of frame_support::traits::Instance
			// This import is not strictly needed but made in order not to have breaking change.
			use #scrate::traits::Instance as #instance_trait;
```

Reply:
What do you mean by local re-export?

Reply:
just a local import: `use #scrate::traits::Instance`

Reply:
what I meant is that in your suggestion you do `use #scrate::traits::Instance as _;`
But this would break module doing just `pub trait Trait<I: Instance = DefaultInstance>`
Instead, if we want not to break, we should do `use #scrate::traits::Instance`

Reply:
Ahh, I'm an idiot xD Yeah that is right :) Sorry. Sounds good to me.